### ISSUE TITLE
Panic when using ScrollView with biding loop on `viewport-height`

### ISSUE THREAD

**Issue by P7XXTMX** [slint-ui/slint#10343]
### Bug Description

Example
```slint
import { ScrollView } from "std-widgets.slint";

export component ScrollableContent inherits Window {
    width: 300px;
    height: 300px;
   ScrollView{
            width: 300px;
            height: 300px;
            viewport-height: rr.height;
         // Your content here
         rr:=Rectangle {
            for i in 10: Text {
                text: "Item " + i;
                y: i * 50px;  
            }  
        }
           
   }
}
```

### Environment Details

- Slint Version:
- Platform/OS:
- Programming Language:
- Backend/Renderer:

**Comment by ogoffart** [comment]
Thanks for reporting the issue. I can reproduce the `Recursion detected` panic.
This seems to be caused because the compiler doesn't detect the binding loop.

**Comment by ogoffart** [comment]
Shorter testcase:
```slint
component ScrollView {
    in-out property <length> viewport-height <=> flickable.viewport-height;
    flickable := Flickable {
        @children
    }
}

export component ScrollableContent inherits Window {
    ScrollView {
        viewport-height: rr.height;
        rr := Rectangle { }
    }
}
```

**Comment by ogoffart** [comment]
Same as https://github.com/slint-ui/slint/issues/4458
I'm closing this issue as duplicate, but thanks anyway for reporting. Feel free to add 👍 on that issue.
